javascript/ql/lib/semmle/javascript/security/dataflow/XssThroughDomCustomizations.qll

```diff
@@ -216,4 +216,30 @@ module XssThroughDom {
       }
     }
   }
+
+
+  /**
+   * Gets a reference to a value obtained by calling `window.getSelection()`. 
+   * https://developer.mozilla.org/en-US/docs/Web/API/Selection
+   */
+  DataFlow::SourceNode getSelectionCall(DataFlow::TypeTracker t) {
+    t.start() and
+    exists(DataFlow::CallNode call |
+      call = DataFlow::globalVarRef("getSelection").getACall()
+    |
+      result = call
+    )
+    or
+    exists(DataFlow::TypeTracker t2 | result = getSelectionCall(t2).track(t2, t))
+  }
+  
+  /**
+   * A source for text from the DOM from a Selection object toString method call
+   * https://developer.mozilla.org/en-US/docs/Web/API/Selection
+   */
```

Comment:
```suggestion
  /**
   * A source for text from the DOM from calling `toString()` on a `Selection` object.
   * The `toString()` method returns the currently selected text in the DOM.
   * https://developer.mozilla.org/en-US/docs/Web/API/Selection
   */
```

I was initially confused by the `toString()` call.   
Because usually `toString()` is used as taint-step, and not a taint-source.   
But in this case using `toString()` as a source is correct.   
So I tried to add a thing to the QLDoc to make that more clear.